### PR TITLE
feat: make invert_diff configurable

### PR DESCRIPTION
**Pull request by eeeXun**
Some user may still need old style of diff. https://github.com/ellisonleao/gruvbox.nvim/discussions/239#discussioncomment-7255063

### CODE REVIEW

**Comment by eeeXun** [comment]
This should close #290

File: lua/gruvbox.lua [hunk 1]
```diff
@@ -360,9 +361,9 @@ local function get_groups()
     PmenuSel = { fg = colors.bg2, bg = colors.blue, bold = config.bold },
     PmenuSbar = { bg = colors.bg2 },
     PmenuThumb = { bg = colors.bg4 },
-    DiffDelete = { bg = colors.dark_red },
-    DiffAdd = { bg = colors.dark_green },
-    DiffChange = { bg = colors.dark_aqua },
+    DiffDelete = config.invert_diff and { fg = colors.bg0, bg = colors.red } or { bg = colors.dark_red },
+    DiffAdd = config.invert_diff and { fg = colors.bg0, bg = colors.green } or { bg = colors.dark_green },
+    DiffChange = config.invert_diff and { fg = colors.bg0, bg = colors.aqua } or { bg = colors.dark_aqua },
```
Comment (ambroisie) [inline lua/gruvbox.lua:366]: Style not quite a revert of the style, if the user has `config.inverse = true`.

```suggestion
    DiffDelete = config.invert_diff and { fg = colors.bg0, bg = colors.red, reverse = config.inverse } or { bg = colors.dark_red },
    DiffAdd = config.invert_diff and { fg = colors.bg0, bg = colors.green, reverse = config.inverse } or { bg = colors.dark_green },
    DiffChange = config.invert_diff and { fg = colors.bg0, bg = colors.aqua, reverse = config.inverse } or { bg = colors.dark_aqua },
```

Comment (ambroisie) [reply lua/gruvbox.lua:366]: Mmh, I think there are actually two things happening in the new configuration:

* The colors changed.
* And the `reverse` is unconditionally false.

I want _both_ the old colours _and_ `reverse = true`.

Whereas you only want the old colours?

**Comment by ellisonleao** [comment]
hey @eeeXun thanks for the PR but this seems like a personal preference choice. Have you tried just using the `overrides` option to get the colors of your preference?

**Comment by eeeXun** [comment]
Hi @ellisonleao, I don't think it's a personal preference. Although I use the new diff style. But there are many sounds https://github.com/ellisonleao/gruvbox.nvim/discussions/239#discussioncomment-7255063 and #290. Not everybody like the new style.
Why not keep the option that user could easily access instead of `overrides` by their own?

File: lua/gruvbox.lua (hunk 1 above)
Comment (eeeXun) [reply lua/gruvbox.lua:366]: You are right

**Comment by ellisonleao** [comment]
> Not everybody like the new style.

It is impossible to please everyone

> Why not keep the option that user could easily access instead of overrides by their own?

because it's a setting for only one highlight which IMO doesn't make sense. Imagine if everyone creates a settings for a small group or single highlights? With the overrides function we can basically give you the ability to use your own personal preference, together with override_colors, if they also don't make sense four your choice.

**Comment by ellisonleao** [comment]
@eeeXun i guess you didn't read the entire sentence, but let me quote it again:

>  Imagine if everyone creates a settings for a small group or single highlights?


> Because not find a good color for new style yet. But one day we may find it.

Excatly. But in the meantime `overrides` is the way to go and not more configs

**Pull request closed without merging**